source/frontend/src/features/admin/financial-codes/list/FinancialCodeListView.tsx

```diff
@@ -0,0 +1,131 @@
+import { Button } from 'components/common/buttons/Button';
+import { TableSort } from 'components/Table/TableSort';
+import { Roles } from 'constants/roles';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import orderBy from 'lodash/orderBy';
+import { Api_FinancialCode } from 'models/api/FinancialCode';
+import moment from 'moment';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { FaPlus } from 'react-icons/fa';
+import { useHistory } from 'react-router';
+import styled from 'styled-components';
+
+import { useFinancialCodeRepository } from '../hooks/useFinancialCodeRepository';
+import FinancialCodeFilter, {
+  defaultFinancialCodeFilter,
+  IFinancialCodeFilter,
+} from './FinancialCodeFilter/FinancialCodeFilter';
+import { FinancialCodeResults } from './FinancialCodeResults/FinancialCodeResults';
+import * as Styled from './styles';
+
+/**
+ * Page that displays acquisition files information.
+ */
+export const FinancialCodeListView: React.FC = () => {
+  const history = useHistory();
+  const { hasRole } = useKeycloakWrapper();
+  const {
+    getFinancialCodes: {
+      execute: getFinancialCodes,
+      response: financialCodeResults,
+      loading: financialCodesLoading,
+    },
+  } = useFinancialCodeRepository();
+
+  const fetchData = useCallback(async () => {
+    await getFinancialCodes();
+  }, [getFinancialCodes]);
+
+  useEffect(() => {
+    if (financialCodeResults === undefined) {
+      fetchData();
+    }
+  }, [fetchData, financialCodeResults]);
+
+  // Sorting and filtering for this list view is performed client-side
+  const [sort, setSort] = React.useState<TableSort<Api_FinancialCode>>({});
+  const [filter, setFilter] = React.useState<IFinancialCodeFilter>(defaultFinancialCodeFilter);
+
+  const sortedFilteredFinancialCodes = useMemo(() => {
+    if (financialCodeResults && financialCodeResults?.length > 0) {
+      let records = [...financialCodeResults];
+      if (filter) {
+        records = records.filter(record => {
+          // Apply all the filters
+          if (filter.financialCodeType && record.type !== filter.financialCodeType) {
+            return false;
+          }
+          if (
+            filter.codeValueOrDescription &&
+            !record.code?.toLowerCase().includes(filter.codeValueOrDescription.toLowerCase()) &&
+            !record.description?.toLowerCase().includes(filter.codeValueOrDescription.toLowerCase())
+          ) {
+            return false;
+          }
+          if (filter.showExpiredCodes === false && isExpiredCode(record)) {
+            return false;
+          }
+          // Finally return true because record has satisfied all conditions
+          return true;
+        });
+      }
+
+      if (sort) {
+        const sortFields = Object.keys(sort);
+        if (sortFields?.length > 0) {
```

Comment:
This doesn't seem to correspond to this requirement: Display the results sorted by Code type, sort order and in absence of sort order Code value